tests/pass/foreign-fn-linkname.rs

```diff
@@ -1,6 +1,3 @@
-//ignore-windows: Uses POSIX APIs
```

Comment:
@oli-obk this seems to have been missed in the `//@` transition. Also turns out the libc crate has enough stuff on Windows to actually make this test work. Still makes me wonder if anything else was missed... (I did some grepping and could not find anything).